http-utils/src/main/java/ai/vespa/util/http/hc5/VespaHttpClientBuilder.java

```diff
@@ -10,47 +11,75 @@
  *
  * @author jonmv
  */
 public class VespaHttpClientBuilder {
+    private HttpClientConnectionManagerFactory connectionManagerFactory = PoolingHttpClientConnectionManager::new;
+    private HostnameVerifier hostnameVerifier = new NoopHostnameVerifier();
+    private boolean rewriteHttpToHttps = true;
+    private final ConnectionConfig.Builder connectionConfigBuilder = ConnectionConfig.custom();
 
     public interface HttpClientConnectionManagerFactory {
-        HttpClientConnectionManager create(Registry<ConnectionSocketFactory> socketFactories);
+        PoolingHttpClientConnectionManager create(Registry<ConnectionSocketFactory> socketFactories);
     }
 
-    public static HttpClientBuilder create() {
-        return create(PoolingHttpClientConnectionManager::new);
+    private VespaHttpClientBuilder() {
     }
 
-    public static HttpClientBuilder create(HttpClientConnectionManagerFactory connectionManagerFactory) {
-        return create(connectionManagerFactory, new NoopHostnameVerifier());
+    public static VespaHttpClientBuilder custom() {
+        return new VespaHttpClientBuilder();
     }
 
-    public static HttpClientBuilder create(HttpClientConnectionManagerFactory connectionManagerFactory,
-                                           HostnameVerifier hostnameVerifier) {
-        return create(connectionManagerFactory, hostnameVerifier, true);
+    public VespaHttpClientBuilder set(HttpClientConnectionManagerFactory connectionManagerFactory) {
+        this.connectionManagerFactory = connectionManagerFactory;
+        return this;
     }
 
-    public static HttpClientBuilder create(HttpClientConnectionManagerFactory connectionManagerFactory,
-                                           HostnameVerifier hostnameVerifier,
-                                           boolean rewriteHttpToHttps) {
+    public VespaHttpClientBuilder set(HostnameVerifier hostnameVerifier) {
+        this.hostnameVerifier = hostnameVerifier;
+        return this;
+    }
+    public VespaHttpClientBuilder setRewriteHttpToHttps(boolean enable) {
+        this.rewriteHttpToHttps = enable;
+        return this;
+    }
+    public VespaHttpClientBuilder setConnectTimeout(long connectTimeout, TimeUnit timeUnit) {
+        connectionConfigBuilder.setConnectTimeout(connectTimeout, timeUnit);
+        return this;
+    }
+    public VespaHttpClientBuilder setConnectTimeout(Timeout connectTimeout) {
+        connectionConfigBuilder.setConnectTimeout(connectTimeout);
+        return this;
+    }
+    public VespaHttpClientBuilder setSocketTimeout(long connectTimeout, TimeUnit timeUnit) {
+        connectionConfigBuilder.setConnectTimeout(connectTimeout, timeUnit);
+        return this;
+    }
+    public VespaHttpClientBuilder setSocketTimeout(Timeout connectTimeout) {
+        connectionConfigBuilder.setConnectTimeout(connectTimeout);
+        return this;
+    }
+
+    public HttpClientBuilder build() {
```

Comment:
Consider adding a `buildClient()`/`build()` that build the client. The `.build().build()` suffix looks a bit silly. 
Then rename previous `build()` to `builder()`/`apacheBuilder()` or something similar.

Reply:
Fixed